core/orchestrator.go

```diff
@@ -753,13 +753,21 @@ func (rtm *RemoteTranscoderManager) Manage(stream net.Transcoder_RegisterTransco
 	rtm.liveTranscoders[transcoder.stream] = transcoder
 	rtm.remoteTranscoders = append(rtm.remoteTranscoders, transcoder)
 	sort.Sort(byLoadFactor(rtm.remoteTranscoders))
+	if monitor.Enabled {
+		monitor.SetTranscodersNumber(int64(len(rtm.liveTranscoders)))
+		monitor.SetTranscodersCapacity(int64(rtm.totalCapacity()))
+	}
 	rtm.RTmutex.Unlock()
 
 	<-transcoder.eof
 	glog.Infof("Got transcoder=%s eof, removing from live transcoders map", from)
 
 	rtm.RTmutex.Lock()
 	delete(rtm.liveTranscoders, transcoder.stream)
+	if monitor.Enabled {
+		monitor.SetTranscodersNumber(int64(len(rtm.liveTranscoders)))
+		monitor.SetTranscodersCapacity(int64(rtm.totalCapacity()))
```

Comment:
Is this "expensive" ? If so, would it make more sense to store these in temp variables and add to Prometheus outside the lock?

Reply:
Don't know, but good idea - moved it out.